### PR TITLE
fix: opening someone's profile menu locks scrollbar and "shifts" header css

### LINKED ISSUE #1696: Bug: opening someone's profile menu locks scrollbar and "shifts" header css
### Describe the bug

https://github.com/open-sauced/app/assets/23109390/2e7a7deb-a973-4758-8563-72bca964e656

2 things with this bug:
1. Clicking the drop down menu (right now used only for following and unfollowing) causes the header CSS to shift over.
2. After investigating abit further, it seems that this "locks" the scrollbar (notice in the video above the scrollbar on the right disappears) and prevents any scrolling

Clicking the dropdown menu should not cause the scrollbar to disappear (or lock scrolling) and shouldn't shift the CSS where it's visible by the user.

This behavior is the same on Chrome and Firefox.

### Steps to reproduce

1. Go to someone's profile: https://insights.opensauced.pizza/user/jpmcb
2. Click the (...) dropdown menu
3. Notice the shift in CSS and lack of being able to scroll

### Browsers

Firefox

### Additional context (Is this in dev or production?)

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by RitaDee**
## Description

This PR addresses the issue of the scrollbar locking and preventing scrolling that was observed in the `ContributorProfileHeader` component. It achieves this by replacing the deprecated `DropdownMenu` component with the `Popover` component from `@radix-ui/react-popover`. Additionally, it replaces the `DropdownMenuItem` components with simple `button` elements and organizes the code by moving the `PopoverContent` component inside the `Popover` component. These changes simplify the code structure and resolve the scrolling issue reported in the issue description.

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Fixes #1696

## Mobile & Desktop Screenshots/Recordings

Before:

https://github.com/open-sauced/app/assets/106407013/7deaeb7b-258b-44a4-9eec-b2d4628ca820

After:

https://github.com/open-sauced/app/assets/106407013/7b1fe39b-8b72-4951-8c82-2b9da7c86f63

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

**Comment by OgDev-01** [comment]
Hey @RitaDee, Thanks for pushing this fix 🍕. Will you take a look at updating the styles of the popover content to match what we have with the dropdown?.

### Expected look
Screenshot 2023-09-18 at 14 19 37

### How it looks now
Screenshot 2023-09-18 at 14 20 56

The width and padding needs a bit of adjustment

**Comment by RitaDee** [comment]
> Hey @RitaDee, Thanks for pushing this fix 🍕. Will you take a look at updating the styles of the popover content to match what we have with the dropdown?.
> 
> ### Expected look
> Screenshot 2023-09-18 at 14 19 37
> 
> ### How it looks now
> Screenshot 2023-09-18 at 14 20 56
> 
> The width and padding needs a bit of adjustment

Sure!

**Review by jpmcb — CHANGES_REQUESTED** [review]
A few things have regressed in this PR:
- The default state of the button should be "Following". Hovering over "Following" should dynamically swap the button over to "Unfollow"
- Hovering over either of the options should highlight it with the orange color
- Sizing of the popover has changed and still isn't quite right

Currently on production:
Screenshot 2023-09-18 at 3 40 24 PM

Proposed change:
Screenshot 2023-09-18 at 3 42 59 PM

**Comment by NsdHSO** [comment]
@RitaDee IMHO in my daily dev  this feature is a little bit annoying because if you scroll the tooltip is above the toolbar, and it is a little bit confusing

**Comment by RitaDee** [comment]
> @RitaDee IMHO in my daily dev this feature is a little bit annoying because if you scroll the tooltip is above the toolbar, and it is a little bit confusing

I have worked on the positioning. That was the fix.

**Comment by ThePiyushAggarwal** [comment]
Some issues that are happening with this change as well.

https://github.com/open-sauced/app/pull/1712#issuecomment-1725812447
https://github.com/open-sauced/app/pull/1712#issuecomment-1725923347

File: components/molecules/ContributorProfileHeader/contributor-profile-header.tsx [hunk 1]
```diff
@@ -182,69 +179,70 @@ const ContributorProfileHeader = ({
...
+                className="flex flex-col gap-1 py-2 px-2 !z-40 w-36 text-sm font-medium rounded-lg bg-white outline-none"
+              >
                 {user ? (
                   !isOwner && (
                     <>
-                      <DropdownMenuItem
+                      <button
                         onClick={handleFollowClick}
-                        className="rounded-md flex items-center gap-1 !cursor-pointer [&>span>span:nth-child(1)]:hover:hidden [&>span>span:nth-child(1)]:focus:hidden [&>span>span:nth-child(2)]:hover:inline [&>span>span:nth-child(2)]:focus:inline"
+                        className="rounded-md flex gap-1 !cursor-pointer hover:text-sauced-orange hover:bg-orange-100 [&>span>span:nth-child(1)]:hover:hidden [&>span>span:nth-child(1)]:focus:hidden [&>span>span:nth-child(2)]:hover:inline [&>span>span:nth-child(2)]:focus:inline"
```
Comment (OgDev-01) [inline components/molecules/ContributorProfileHeader/contributor-profile-header.tsx:197]: having this odd behavior when i try to hover by the end right character 
Screenshot 2023-09-19 at 18 39 00
```suggestion
                        className="rounded-md flex gap-1 !cursor-pointer hover:text-sauced-orange hover:bg-orange-100"
```
Those pseudo-classes are not needed anymore.

File: components/molecules/ContributorProfileHeader/contributor-profile-header.tsx [hunk 2]
```diff
@@ -182,69 +179,70 @@ const ContributorProfileHeader = ({
...
                       >
                         {/* `span` tag changes below must be in line with the styles on the parent */}
                         <span className="pl-3 pr-7">
                           {isFollowing ? (
-                            <>
-                              <span className="">Following</span>
-                              <span className="hidden">Unfollow</span>
-                            </>
+                            <div className="group relative">
```
Comment (OgDev-01) [inline components/molecules/ContributorProfileHeader/contributor-profile-header.tsx:202]:
```suggestion
                            <>
```
Keep this as a fragment so the hover effect works on not just the text but the fill width of the button

https://github.com/open-sauced/app/assets/62995161/1c1d8378-4819-48dc-be20-dade8874aea8

File: components/molecules/ContributorProfileHeader/contributor-profile-header.tsx [hunk 3]
```diff
@@ -182,69 +179,70 @@ const ContributorProfileHeader = ({
...
                   !isOwner && (
                     <>
-                      <DropdownMenuItem
+                      <button
                         onClick={handleFollowClick}
-                        className="rounded-md flex items-center gap-1 !cursor-pointer [&>span>span:nth-child(1)]:hover:hidden [&>span>span:nth-child(1)]:focus:hidden [&>span>span:nth-child(2)]:hover:inline [&>span>span:nth-child(2)]:focus:inline"
+                        className="rounded-md flex gap-1 !cursor-pointer hover:text-sauced-orange hover:bg-orange-100 [&>span>span:nth-child(1)]:hover:hidden [&>span>span:nth-child(1)]:focus:hidden [&>span>span:nth-child(2)]:hover:inline [&>span>span:nth-child(2)]:focus:inline"
                       >
                         {/* `span` tag changes below must be in line with the styles on the parent */}
                         <span className="pl-3 pr-7">
```
Comment (OgDev-01) [inline components/molecules/ContributorProfileHeader/contributor-profile-header.tsx:200]: Move the `group` classname to this span and also make it span the full with of the button.

**Review by OgDev-01 — CHANGES_REQUESTED** [review]
We are almost there. Just a few tweaks and this is good to go 👍

**Comment by ThePiyushAggarwal** [comment]
Please check this out. https://github.com/open-sauced/app/pull/1712#issuecomment-1726300898

**Comment by RitaDee** [comment]
> Please check this out. [#1712 (comment)](https://github.com/open-sauced/app/pull/1712#issuecomment-1726300898)

Thank you @ThePiyushAggarwal. I have done the fix.

**Review by jpmcb — APPROVED** [review]
Great work on this!!!!! And shoutout to @ThePiyushAggarwal for the great investigation / deep dive on this too 🚀 🍕

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**